lib/rust/ensogl/core/src/display/camera/camera2d.rs

```diff
@@ -425,7 +428,14 @@ impl Camera2d {
 
     /// Update all dirty camera parameters and compute updated view-projection matrix.
     pub fn update(&self, scene: &Scene) -> bool {
-        self.data.borrow_mut().update(scene)
+        let is_updated = self.data.borrow_mut().update(scene);
+        if is_updated {
+            let borrowed = self.data.borrow();
+            let frp = &borrowed.frp;
+            frp.position.emit(self.display_object.position());
+            frp.zoom.emit(borrowed.zoom);
```

Comment:
Here also we still borrow data while emitting events.

Reply:
Fixed